DSharpPlus.UnifiedCommands/Internals/CommandModuleRegister.cs

```diff
@@ -0,0 +1,344 @@
+        Assembly assembly,
+        DiscordClient client)
+    {
+        IEnumerable<Type> classes = assembly.GetTypes()
+            .Where(t => t.IsPublic &&
+                        t.IsClass &&
+                        !t.IsAbstract &&
+                        t.IsSubclassOf(typeof(ApplicationModule)) &&
+                        t.GetCustomAttribute<ApplicationModuleAttribute>() is not null);
+
+        NullabilityInfoContext nullabilityContext = new();
+
+        List<DiscordApplicationCommand> commands = new();
+        foreach (Type @class in classes)
+        {
+            ObjectFactory objectFactory = ActivatorUtilities.CreateFactory(@class, Array.Empty<Type>());
+            ApplicationModuleData moduleData = new(objectFactory);
+
+            bool registerAsSubcommands = false;
+            ApplicationModuleAttribute? moduleAttribute = @class.GetCustomAttribute<ApplicationModuleAttribute>();
+            if (moduleAttribute?.Name is not null)
+            {
+                registerAsSubcommands = true;
+            }
+
+            DiscordApplicationCommandBuilder? commandBuilder = null;
+            if (registerAsSubcommands)
+            {
+                commandBuilder = new();
+                commandBuilder.WithName(moduleAttribute!.Name!).WithDescription(moduleAttribute.Description!)
+                    .WithType(ApplicationCommandType.SlashCommand);
+            }
+
+            foreach (MethodInfo method in @class.GetMethods())
+            {
+                ApplicationNameAttribute? attribute = method.GetCustomAttribute<ApplicationNameAttribute>();
+                if (attribute is null)
+                {
+                    continue;
+                }
+
+                string applicationName = registerAsSubcommands ? $"{moduleAttribute!.Name} {attribute.Name}" : attribute.Name;
+                DiscordApplicationCommandOptionBuilder? option = null;
+                if (!registerAsSubcommands)
+                {
+                    commandBuilder = new();
+                    commandBuilder.WithType(ApplicationCommandType.SlashCommand).WithName(attribute.Name)
+                        .WithDescription(attribute.Description);
+                }
+                else
+                {
+                    option = new();
+                    option.WithName(attribute.Name).WithDescription(attribute.Description)
+                        .WithType(ApplicationCommandOptionType.SubCommand);
+                }
+
+                List<ApplicationMethodParameterData> parameters = new();
+                foreach (ParameterInfo parameter in method.GetParameters())
+                {
+                    ApplicationOptionAttribute? name = parameter.GetCustomAttribute<ApplicationOptionAttribute>();
+                    if (name is null)
+                    {
+                        throw new Exception("Parameter needs to have `ApplicationNameAttribute` marked.");
```

Comment:
Would be nice if the error message contains the methodname, so the user can see easily where they have to change something